src/python/WMComponent/RucioInjector/RucioInjectorPoller.py

```diff
@@ -0,0 +1,426 @@
+        else:
+            if recovery:
+                self.filesToRecover = []
+
+    def closeBlocks(self):
+        """
+        Close any blocks that have been migrated to global DBS
+        """
+        logging.info("Starting closeBlocks method")
+
+        # in short, dbsbuffer_file.in_phedex = 1 AND dbsbuffer_block.status = 'InDBS'
+        migratedBlocks = self.getMigrated.execute()
+        ### FIXME the data format returned by this DAO
+        for location in migratedBlocks:
+            for container in migratedBlocks[location]:
+                for block in migratedBlocks[location][container]:
+                    if self.rucio.closeBlockContainer(block):
+                        self.setBlockClosed.execute(block)
+                    else:
+                        logging.error("Failed to close block: %s. Will retry again later.", block)
+
+    def deleteBlocks(self):
+        """
+        _deleteBlocks_
+        Find deletable blocks, then decide if to delete based on:
+        Is there an active subscription for dataset or block ?
+          If yes => set deleted=2
+          If no => next check
+        Has transfer to all destinations finished ?
+          If yes => request block deletion, approve request, set deleted=1
+          If no => do nothing (check again next cycle)
+        """
+        raise NotImplementedError
+
+    def deleteBlocksPhEDExCalls(self, location, blocksToDelete):
+        """
+        _deleteBlocksPhEDExCalls_
+        actual PhEDEx calls for block deletion
+        """
+        deletion = PhEDExDeletion(blocksToDelete.keys(), location,
+                                  level='block',
+                                  comments="WMAgent blocks auto-delete from %s" % location,
+                                  blocks=blocksToDelete)
+
+        try:
+            response = self.phedex.delete(deletion)
+            requestId = response['phedex']['request_created'][0]['id']
+            # auto-approve deletion request
+            self.phedex.updateRequest(requestId, 'approve', location)
+        except Exception as ex:
+            logging.error("PhEDEx block delete/approval failed with Exception: %s", str(ex))
+            logging.debug("Traceback: %s", str(traceback.format_exc()))
+        else:
+            binds = []
+            for dataset in blocksToDelete:
+                for blockName in blocksToDelete[dataset]:
+                    binds.append({'DELETED': 1,
+                                  'BLOCKNAME': blockName})
+            self.markBlocksDeleted.execute(binds)
+
+        return
+
+    # TODO: this will likely go away once the phedex to rucio migration is over
+    def _isContainerTierAllowed(self, dsetName):
+        """
+        Check the container datatier against the list of data tiers that we want
+        to get pinned in Rucio through a container rule
+        :return: boolean whether we can or not insert a rule
+        """
+        for skipTier in self.skipRulesForTiers:
```

Comment:
Same comment, more idiomatic python could be used.

Reply:
thanks!